### ISSUE TITLE
Bug in rustdoc test runner or am I braindead?

### ISSUE THREAD

**Issue by tcharding** [rust-bitcoin/rust-secp256k1#508]
In `lib.rs` we have code in rustdoc that should not compile but it does
```rust
//! If the "global-context" feature is enabled you have access to an alternate API.
//!
//! ```rust
//! # #[cfg(all(feature="global-context", feature = "std", feature="rand-std", features = "bitcoin_hashes"))] {
//! use secp256k1::rand::thread_rng;
//! use secp256k1::{generate_keypair, Message};
//! use secp256k1::hashes::sha256;
//!
//! let (secret_key, public_key) = generate_keypair(&mut thread_rng());
//! let message = Message::from_hashed_data::<sha256::Hash>("Hello World!".as_bytes());
//!
//! let sig = secret_key.sign_ecdsa(&secret_key, &message);
//! assert!(sig.verify(&message, &public_key).is_ok());
//! # }
//! ```
```

I ran this code with: `cargo test --doc --features=std,rand-std,bitcoin-hashes-std,global-context`. Its the call to `sign_ecdsa` which is broken and indeed adding some garbage on the end of the line makes the test fail.

To try to prove that I'm not being brain dead I wrote

```rust
    #[test]
    fn wtf() {
        use crate::rand::thread_rng;
        use crate::{generate_keypair, Message};
        use crate::hashes::sha256;
        let (secret_key, public_key) = generate_keypair(&mut thread_rng());
        let message = Message::from_hashed_data::<sha256::Hash>("Hello World!".as_bytes());
        let sig = secret_key.sign_ecdsa(&message, &secret_key);
        assert!(sig.verify(&message, &public_key).is_ok());
    }
```
And ran it with: `cargo test --doc --features=std,rand-std,bitcoin-hashes-std,global-context wtf`.

**Comment by apoelstra** [comment]
If you remove `--doc` from your second `cargo` command then you get a compilation failure in the `wtf` unit test as expected.

As for why the doctest is not run as part of the docs, I suspect it is because one of the `feature=` conditions has `features` (note the s) instead of `feature`. Disappointing/frustrating that no tooling complains about this.

**Comment by tcharding** [comment]
> As for why the doctest is not run as part of the docs, I suspect it is because one of the feature= conditions has features (note the s) instead of feature. Disappointing/frustrating that no tooling complains about this.

Ha! Yeah writing rustdoc examples feels like writing C with the nano editor.

**Comment by tcharding** [comment]
Closed by #510
